### DISCUSSION TITLE
Erro no ISAPI do Horse no IIS (Erro 405)

### DISCUSSION THREAD

**Discussion by felipelavor73** [HashLoad/horse#298] [General]
O método GET Ping/Pong está funcionando normalmente, mas o método abaixo que criei, sempre dá o erro 405 - O verbo HTTP usado para acessar esta página não é permitido. 

procedure Registry;
begin
    THorse.Post('/usuario/auth', Auth);
end;

Coloco na chamada POST->http://site.com.br/site.dll/usuario/auth

Testei com e sem Rewrite URL e mesmo assim não vai. 

Alguém alguma luz ? 

Só para complementar, o erro: Method Not Allowed. Alguma forma de ativar o POST no IIS ?

**Comment by alcantarajoaopaulo** [comment]
Pode estar bloqueado no próprio servidor da aplicação.
Verifique dentro do ISS:
- Escolhe seu site no menu à esquerda
- Clique em Mapeamento de Módulos
- Abra o recurso ISAPI
- Clique em Restrições da Solicitação
- Clique na guia Verbos
- Verifique qual opção está escolhida (Todos os verbos, somente GET)
- Reconfigure, salve, e reinicie seu site para testar novamente

**Reply by felipelavor73** [reply]
Obrigado pelo retorno. Fiz isso, ainda deu erro, mas vi que precisava por a dll do mysql no syswow64 e também configurar pra SSL porque vem o domínio pela cloudflare. 

Valeu pela ajuda.

**Reply by viniciussanchez** [reply]
@felipelavor73 tente fazer essa configuração, e veja se o erro muda. 

![image](https://user-images.githubusercontent.com/16382981/186012938-4e61d758-8151-4a1d-bf57-7cfefec85986.png)

**Reply by viniciussanchez** [reply]
https://wildpartyofficial.com/how-do-i-fix-error-405-method-not-allowed-in-iis/
https://docs.microsoft.com/en-us/troubleshoot/developer/webapps/iis/health-diagnostic-performance/http-error-405-website
https://ignas.me/tech/405-method-not-allowed-iis/
